app/models/ems_refresh/save_inventory_infra.rb

```diff
@@ -159,6 +159,10 @@ def save_hosts_inventory(ems, hashes, target = nil)
           #   and the new hostname is not an ip address
           h[:name] = h[:hostname] if h[:name] =~ ip_part && !(h[:hostname] =~ ip_whole)
 
+          # Update the name to be found.name if it has the same ems_ref and the name
+          # already has a '- int' suffix to work around duplicate hostnames
+          h[:name] = found.name if found.name =~ /#{h[:name]} - \d+$/
```

Comment:
Those two if conditionals can't overlap...i.e. you'll do one or the other or neither, but never both.  So, collapsing into the if/elsif makes that fact much clearer.  If you hit the first part, then that means the existing name looks like an ip address, so we are replacing with the hostname.  If you happen to do that, I don't think the hostname (which has now become the name) can *ever* look like `"localhost - 2"`